### ISSUE TITLE
Add encryption for file names and contents

### ISSUE THREAD

**Issue by OmlineEditor** [phpbg/easysync#29]
please add the ability to encrypt synchronized files. I need to upload data from my phone to the server, but I don't trust the server and I want to protect the data. if you encrypt the file name and its contents, then there will be no problems.

To encrypt the file name and contents, you can use a simple library for the AES algorithm.
the file name will be additionally [base58 ](https://en.wikipedia.org/wiki/Binary-to-text_encoding#Base58)converted

**Comment by phpbg** [comment]
I like the idea, but this is not my priority right now.

Let's keep this open and see if people are interested in it.

**Comment by kaoneko** [comment]
Yes, client side encryption is always a good idea when storing your data on someone else's computer (colloquially known as "the cloud").

If you ever decide to implement this, you might want to have a look at how Rclone implements this: 
https://github.com/rclone/rclone/tree/master/backend%2Fcrypt

Or how other apps use rclone behind the scenes:
https://github.com/koofr/vault
https://github.com/newhinton/Round-Sync

Good luck with your app!

**Comment by jnnkB** [comment]
I really like this app, but if it could encrypt the data, it would be perfect :)

**Comment by lessload** [comment]
i will use this app in real life, if it have this feature.
